test/functional/pdo_sqlsrv/pdo_ae_output_param_decimals.phpt

```diff
@@ -0,0 +1,241 @@
+--TEST--
+Test for retrieving encrypted data of decimals/numerics as output parameters
+--DESCRIPTION--
+Use PDOstatement::bindParam with all PDO::PARAM_ types
+--ENV--
+PHPT_EXEC=true
+--SKIPIF--
+<?php require('skipif_mid-refactor.inc'); ?>
+--FILE--
+<?php
+require_once("MsCommon_mid-refactor.inc");
+
+$dataTypes = array("decimal", "numeric");
+$precisions = array(1 => array(0, 1), 
+                    4 => array(0, 1, 4),
+                    16 => array(0, 1, 4, 16),
+                    38 => array(0, 1, 4, 16, 38));
+$inputValuesInit = array(92233720368547758089223372036854775808, -92233720368547758089223372036854775808);
+$inputPrecision = 38;
+
+$errors = array("IMSSP" => "An invalid PHP type was specified as an output parameter. DateTime objects, NULL values, and streams cannot be specified as output parameters.");
+
+$pdoParamTypes = array(
+    PDO::PARAM_BOOL,    // 5
+    PDO::PARAM_NULL,    // 0
+    PDO::PARAM_INT,     // 1 
+    PDO::PARAM_STR,     // 2
+    PDO::PARAM_LOB      // 3
+);
+
+function printValues($msg, $det, $rand, $inputValues)
+{
+    echo $msg;
+    echo "input 0: "; var_dump($inputValues[0]);
+    echo "fetched: "; var_dump($det);
+    echo "input 1: "; var_dump($inputValues[1]);
+    echo "fetched: "; var_dump($rand);
+}
+
+// this function returns true if the floats are more different than expected
+function compareFloats($actual, $expected) 
+{
+    $epsilon = 0.00001;
+    $diff = abs(($actual - $expected) / $expected);
+    return ($diff > $epsilon);
+}
+
+// function compareDecimals() returns false when the fetched values 
+// are different from the expected inputs 
+function compareIntegers($det, $rand, $inputValues, $pdoParamType)
+{
+    ///////////////////////////////////////////////////////////////////////
+    // See GitHub issue 707 - Fix this method when the problem is addressed
+    //
+    // Assume $pdoParamType is PDO::PARAM_BOOL or PDO::PARAM_INT
+    if (is_string($det)) {
+        return (!compareFloats(floatval($det), $inputValues[0]) 
+                && !compareFloats(floatval($rand), $inputValues[1]));

```

Comment:
when would $det be a string when you specify as PARAM_BOOL or PARAM_INT? so it returns a string with decimal places?